File: dataGovern/metadata/atlas-versions/atlas-3.0.0-SNAPSHOT/atlas-examples/sample-app/src/main/java/org/apache/atlas/examples/sampleapp/DiscoveryExample.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.atlas.examples.sampleapp;

import org.apache.atlas.AtlasClientV2;
import org.apache.atlas.AtlasServiceException;
import org.apache.atlas.model.discovery.AtlasQuickSearchResult;
import org.apache.atlas.model.discovery.AtlasSearchResult;
import org.apache.atlas.model.instance.AtlasEntityHeader;

import java.util.List;

public class DiscoveryExample {
    private static final String[] DSL_QUERIES = new String[] { "from DataSet", "from Process" };

    private final AtlasClientV2 client;

    DiscoveryExample(AtlasClientV2 client) {
        this.client = client;
    }

    public void testSearch() {
        for (String dslQuery : DSL_QUERIES) {
            try {
                AtlasSearchResult       result      = client.dslSearchWithParams(dslQuery, 10, 0);
                List<AtlasEntityHeader> entities    = result != null ? result.getEntities() : null;
                int                     resultCount = entities == null ? 0 : entities.size();

                SampleApp.log("DSL Query: " + dslQuery);
                SampleApp.log("  result count: " + resultCount);

                for (int i = 0; i < resultCount; i++) {
                    SampleApp.log("  result # " + (i + 1) + ": " + entities.get(i));
                }
            } catch (Exception e) {
                SampleApp.log("query -: " + dslQuery + " failed");
            }
        }
    }

    public void quickSearch(String searchString) {
        try {
            AtlasQuickSearchResult  result      = client.quickSearch(searchString, SampleAppConstants.TABLE_TYPE, false, 2, 0);
            List<AtlasEntityHeader> entities    = result != null && result.getSearchResults() != null ? result.getSearchResults().getEntities() : null;
            int                     resultCount = entities == null ? 0 : entities.size();

            SampleApp.log("Quick search: query-string=" + searchString);
            SampleApp.log("  result count: " + resultCount);

            for (int i = 0; i < resultCount; i++) {
                SampleApp.log("  result # " + (i + 1) + ": " + entities.get(i));
            }
        } catch (AtlasServiceException e) {
            e.printStackTrace();
        }
    }

    public void basicSearch(String typeName, String classification, String query) {
        try {
            AtlasSearchResult       result      = client.basicSearch(typeName, classification, query, false, 2, 0);
            List<AtlasEntityHeader> entities    = result != null ? result.getEntities() : null;
            int                     resultCount = entities == null ? 0 : entities.size();

            SampleApp.log("Basic search: typeName=" + typeName + ", classification=" + classification + ", query=" + query);
            SampleApp.log("  result count: " + resultCount);

            for (int i = 0; i < resultCount; i++) {
                SampleApp.log("  result # " + (i + 1) + ": " + entities.get(i));
            }
        } catch (AtlasServiceException e) {
            e.printStackTrace();
        }
    }
}

File: govern/meta-data/atlas-release-2.3.0/atlas-examples/sample-app/src/main/python/discovery_example.py
#!/usr/bin/env/python

#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import logging

from utils import TABLE_TYPE

LOG = logging.getLogger('discovery-example')


class DiscoveryExample:
    DSL_QUERIES = {"from DataSet", "from Process"}

    def __init__(self, client):
        self.typesDef = None
        self.client   = client

    def dsl_search(self):
        for dsl_query in DiscoveryExample.DSL_QUERIES:
            try:
                result = self.client.discovery.dsl_search_with_params(dsl_query, 10, 0)

                if result:
                    entities_result = result.entities

                    if entities_result:
                        LOG.info("query: '%s' retrieved: %s rows", dsl_query, len(entities_result))

            except Exception as e:
                LOG.exception("query: %s failed in dsl search", dsl_query)

    def quick_search(self, search_string):
        try:
            result = self.client.discovery.quick_search(search_string, TABLE_TYPE, False, 2, 0)

            if result:
                LOG.info("Quick-search result: %s", result.searchResults)

        except Exception as e:
            LOG.exception("query: '%s' failed in quick search", search_string)

    def basic_search(self, type_name, classification, query):
        try:
            result = self.client.discovery.basic_search(type_name, classification, query, False, None, 'ASCENDING', 2, 0)

            if result:
                LOG.info("Basic-search result: %s", result)

        except Exception as e:
            LOG.exception("query: '%s' failed in basic search", query)
